elasticsearch_dsl/document_base.py

```diff
@@ -28,26 +42,174 @@ def __init__(self, *args, **kwargs):
         self.args, self.kwargs = args, kwargs
 
 
+class InstrumentedField:
+    """Proxy object for a mapped document field.
+
+    An object of this instance is returned when a field is access as a class
+    attribute of a ``Document`` or ``InnerDoc`` subclass. These objects can
+    be used in any situation in which a reference to a field is required, such
+    as when specifying sort options in a search::
+
+        class MyDocument(Document):
+            name: str
+
+        s = MyDocument.search()
+        s = s.sort(-MyDocument.name)  # sort by name in descending order
+    """
+
+    def __init__(self, name, field):
+        self._name = name
+        self._field = field
+
+    def __getattr__(self, attr):
+        f = None
+        try:
+            f = self._field[attr]
+        except KeyError:
+            pass
+        if isinstance(f, Field):
+            return InstrumentedField(f"{self._name}.{attr}", f)
+        return getattr(self._field, attr)
```

Comment:
`getattr(self._field, attr)` is equivalent to `self._field[attr]`. Did you intend to specify None as a default?

Maybe we should add https://github.com/PyCQA/flake8-bugbear which would have caught it (rule B009)

Reply:
You are right, I already have the attribute in `f`, so here I just need to return it.